site/content/docs/main/restore-reference.md

```diff
@@ -36,57 +21,129 @@ Available Commands:
+    * If the resource is a Custom Resource Definition, the `RestoreController` waits for its availability in the cluster. The timeout is 1 minute.
+
+    If any failures happen finishing these steps, the `RestoreController` will log an error in the restore result and will continue restoring.
+
+## Restore order
+
+By default, Velero will restore resources in the following order:
+
+* Custom Resource Definitions
+* Mamespaces
+* StorageClasses
+* VolumeSnapshotClass
+* VolumeSnapshotContents
+* VolumeSnapshots
+* PersistentVolumes
+* PersistentVolumeClaims
+* Secrets
+* ConfigMaps
+* ServiceAccounts
+* LimitRanges
+* Pods
+* ReplicaSets
+* Clusters
+* ClusterResourceSets
+
+It's recommended that you use the default order for your restores. You are able to customize this order if you need to by setting the `--restore-resource-priorities` flag on the Velero server and specifying a different resource order. This customized order will apply to all future restores. You don't have to specify all resources in the `--restore-resource-priorities` flag. Velero will append resources not listed to the end of your customized list in alphabetical order.
+
+```shell
+velero server \
+--restore-resource-priorities=customresourcedefinitions,namespaces,storageclasses,\
+volumesnapshotclass.snapshot.storage.k8s.io,volumesnapshotcontents.snapshot.storage.k8s.io,\
+volumesnapshots.snapshot.storage.k8s.io,persistentvolumes,persistentvolumeclaims,secrets,\
+configmaps,serviceaccounts,limitranges,pods,replicasets.apps,clusters.cluster.x-k8s.io,\
+clusterresourcesets.addons.cluster.x-k8s.io
 ```
 
-The clusters which are not explicitly specifying nodePorts still may need to restore original NodePorts in case of disaster. Auto assigned nodePorts most probably defined on Load Balancers which located front side of cluster.  Changing all these nodePorts on Load Balancers is another operation complexity after disaster if nodePorts are changed.
 
-Velero has a flag to let user deciding the preservation of nodePorts. **`velero restore create`** sub command has  **`--preserve-nodeports`** flag to **preserve** Service nodePorts **always** regardless of nodePorts **explicitly specified** or **not**. This flag used for preserving the original nodePorts from backup and can be used as **`--preserve-nodeports`** or **`--preserve-nodeports=true`**
+## Restoring Persistent Volumes and Persistent Volume Claims
 
-If this flag given and/or set to true, Velero does not remove the nodePorts when restoring Service and tries to use the nodePorts which written on backup.
+Velero has three approaches when restoring a PV, depending on how the backup was taken.
 
-Trying to preserve nodePorts may cause **port conflicts** when restoring on situations below:
+1. When restoring a snapshot, Velero statically creates the PV and then binds it to a restored PVC. Velero's PV rename and remap process is used only in this case because this is the only case where Velero creates the PV resource directly.
```

Comment:
I'm not sure about this. For none Restic or CSI case, the PV restore is done in plugin.
Restoring from VolumeSnapshot needs a snapshot created and recorded in backup, but AFAIK, VolumeSnapshot is only created in CSI backup process.